mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -3630,13 +3651,31 @@ func expandMonitoringConfig(configured interface{}) *container.MonitoringConfig
 	if len(l) == 0 || l[0] == nil {
 		return nil
 	}
-
+<% if version == 'ga' -%>
 	config := l[0].(map[string]interface{})
 	return &container.MonitoringConfig{
 		ComponentConfig: &container.MonitoringComponentConfig{
 			EnableComponents: convertStringArr(config["enable_components"].([]interface{})),
 		},
 	}
+<% end -%>
+<% if version == 'beta' -%>
+	mc := &container.MonitoringConfig{}
+	config := l[0].(map[string]interface{})
+	if v, ok := config["enable_components"]; ok && len(v.([]interface{})) > 0 {
+		enable_components := v.([]interface{})[0].([]interface{})
+		mc.ComponentConfig = &container.MonitoringComponentConfig{
+			EnableComponents: convertStringArr(enable_components),
+		}
+	}
+	if v, ok := config["managed_prometheus"]; ok && len(v.([]interface{})) > 0 {
```

Comment:
It seems like to make this simpler, the managed_prometheus handling could be the only thing behind the `<% if version == 'beta' %>`.  I think that would leave the behavior of the ComponentConfig the same, since it should always be specified in GA.

Reply:
Done.